### PR TITLE
Dasd 8851: Added separate Ubuntu 18, Ubuntu 20 build agents

### PR DESCRIPTION
**Pull request by jack-cloud-platform**
Dasd 8851: Added separate Ubuntu 18, Ubuntu 20 build agents
The two sets of agents support different .NET Core versions, with a corresponding agent capability of LATEST_DOTNET_VERSION:
- Ubuntu 18 agents support .NET Core 2.2
- Ubuntu 20 agents support .NET Core 3.1

### CODE REVIEW

File: Linux/Ubuntu-20/Base/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+RUN apt-get update \
+&& apt-get install -y --no-install-recommends \
+        ca-certificates \
+        curl \
+        jq \
+        git \
+        iputils-ping \
+        libcurl4 \
+        libunwind8 \
```
Comment (nbowes24) [inline Linux/Ubuntu-20/Base/Dockerfile:16]: Do we know that this doesn't need libicu as per docs & ubuntu 18?

File: Linux/Ubuntu-20/Base/azure-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    
+  - script: sudo apt-get install azure-cli
+  
+  - template: azure-pipelines-templates/build/step/dockerfile-build.yml@das-platform-building-blocks
+    parameters:
+      ContainerRegistryName: $(PublicAcrName)
+      ServiceConnection: SFA-DIG-Prod-ARM
+      ImageName: azure-pipelines-base-build-agent-ubuntu-20
+      WorkingDirectory: Linux/Ubuntu-20/Base
+      BranchToCreateLatestTag: DASD-8851-ubuntu-18-20-build-agents.2
```
Comment (nbowes24) [inline Linux/Ubuntu-20/Base/azure-pipelines.yml]: Reminder to change this after ✔️ testing

File: Linux/Ubuntu-18/Build/manifest.yml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+                labelSelector:
+                  matchExpressions:
+                    - key: app
+                      operator: In
+                      values:
+                        - azure-pipelines-build-agent-ubuntu-18
+                topologyKey: failure-domain.beta.kubernetes.io/zone
+      containers:
+        - name: azure-pipelines-build-agent-ubuntu-18
+          image: __PublicAcrName__.azurecr.io/azure-pipelines-build-agent-ubuntu-18:__ImageVersion__
```
Comment (nbowes24) [inline Linux/Ubuntu-18/Build/manifest.yml:39]: Maybe now is a good time to hardcode latest?

File: Linux/Ubuntu-20/Build/manifest.yml [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+                labelSelector:
+                  matchExpressions:
+                    - key: app
+                      operator: In
+                      values:
+                        - azure-pipelines-build-agent-ubuntu-20
+                topologyKey: failure-domain.beta.kubernetes.io/zone
+      containers:
+        - name: azure-pipelines-build-agent-ubuntu-20
+          image: __PublicAcrName__.azurecr.io/azure-pipelines-build-agent-ubuntu-20:__ImageVersion__
```
Comment (nbowes24) [inline Linux/Ubuntu-20/Build/manifest.yml:39]: Maybe now is a good time to hardcode latest?

File: Linux/Ubuntu-18/Base/azure-pipelines.yml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    ref: refs/tags/0.3.23
+    endpoint: SFA
+
+pool:
+  vmImage: ubuntu-latest
+
+variables:
+- group : Release Management Resources
+
+steps:
```
Comment (nbowes24) [inline Linux/Ubuntu-18/Base/azure-pipelines.yml:36]: Missing the comment here

File: Linux/Ubuntu-18/Build/manifest.yml (hunk 3 above)
Comment (jack-cloud-platform) [reply Linux/Ubuntu-18/Build/manifest.yml:39]: I think handy to have flexibility of the release using build's version.

File: Linux/Ubuntu-20/Build/manifest.yml (hunk 4 above)
Comment (jack-cloud-platform) [reply Linux/Ubuntu-20/Build/manifest.yml:39]: I think handy to have flexibility of the release using build's version.

File: Linux/Ubuntu-20/Base/Dockerfile (hunk 1 above)
Comment (jack-cloud-platform) [reply Linux/Ubuntu-20/Base/Dockerfile]: Added libicu66 as supported on Ubuntu 20

**Review by nbowes24 — APPROVED** [review]

**Pull request merged**